controllers/external/tracker.go

```diff
@@ -42,7 +41,7 @@ type ObjectTracker struct {
 }
 
 // Watch uses the controller to issue a Watch only if the object hasn't been seen before.
-func (o *ObjectTracker) Watch(log logr.Logger, obj runtime.Object, handler handler.EventHandler, p ...predicate.Predicate) error {
+func (o *ObjectTracker) Watch(log logr.Logger, obj client.Object, handler handler.TypedEventHandler[client.Object], p ...predicate.TypedPredicate[client.Object]) error {
```

Comment:
Q: Could this diff be even smaller by keep using Predicate etc.? (type Predicate = TypedPredicate[client.Object])